chapters/annotations.tex

```diff
@@ -68,27 +73,190 @@ \section{Annotations for Documentation}\doublelabel{annotations-for-documentatio
+  String identifier "Identifier meant for programmatic access";
+  String group = "" "Name of plot group";
+  Boolean preferred = false "Automatically display figure after simulation";
+  Plot[:] plots "Plots";
+  String caption "Figure caption";
+end Figure;
+\end{lstlisting}
+
+A \lstinline!Figure! is a graphical container that can contain several
+\lstinline!plots! described by \lstinline!Plot! annotations:
+\begin{lstlisting}[language=modelica]
+record Plot
+  String title "Title meant for display";
+  String identifier "Identifier meant for programmatic access";
+  Curve[:] curves "Plot curves";
+  Axis x "X axis properties";
+  Axis y "Y axis properties";
+end Plot;
+\end{lstlisting}
+
+A \lstinline!Plot! can contain several \lstinline!curves!, see
+\autoref{plot-curves}, that all share a common x and y axis with properties
+described in \autoref{axis-properties}.
+
+The \lstinline!title! of a \lstinline!Figure! shall be non-empty and is
+mandatory, as every figure needs a title for presentation in contexts such as a
+list of figures, and as it is generally hard for tools to automatically generate
+a meaningful title.  On the other hand, when the \lstinline!title! of a
+\lstinline!Plot! isn't provided, the tool produces a default, but the default is
+allowed to be empty.  Providing the empty string as \lstinline!title! of a
+\lstinline!Plot! means that no title should be shown.  The plot title is not to
+be confused with the plot \emph{label} which is never empty, see below. Variable
+replacements, as described in \autoref{variable-replacements}, can be used in the
+\lstinline!title! of \lstinline!Figure! and \lstinline!Plot!.
+
+The \lstinline!identifier! in \lstinline!Figure! and \lstinline!Plot! is an
+optional \lstinline!String! identifier, and is intended to identify the
+\lstinline!Figure! and \lstinline!Plot! for programmatic access.  A
+\lstinline!Figure! must be uniquely identified by its \lstinline!identifier! and
+containing class.  This means that a \lstinline!Figure! \lstinline!identifier!
+must be unique among all \lstinline!Figure! annotations within the same
+\lstinline!figures! annotation as well as among all \lstinline!figures!
+annotations from inherited classes.  A \lstinline!Plot! \lstinline!identifier!
+on the otherhand is only required to be unique among the \lstinline!plots! in
+the the same \lstinline!Figure! annotation.
+
+\begin{nonnormative}
+For \lstinline!Figure!, this makes it possible to reference the plot from a
+tool-specific scripting environment. For \lstinline!Plot!, this makes it
+possible to reference the plot in the figure caption, which becomes useful when
+the \lstinline!Figure! contains more than one \lstinline!Plot!.
+\end{nonnormative}
+
+Every \lstinline!Plot! has an automatically generated \emph{label} which is
+required to be shown as soon as at least one \lstinline!Plot! in the
+\lstinline!Figure! has an \lstinline!identifier!.  A tool is free to choose both
```

Comment:
It was a deliberate design choice to make the visual appearance depend in a simple fashion on the presence of `identifier`.  Two use cases supporting this choice (assuming the plot isn't referenced from the `caption`):
- One might want to refer to the plot from a different figure or the `documentation` annotation, perhaps even from a different package.  As long as the tool is aware of the plot being referenced, it should be able to provide a nicely formatted hyper-link to that plot.  Currently, there is no standardized way of doing this, but there is MCP0037 _Generalized Modelica URIs for external resources_ that should address this need.
- The figure will be referenced from a different context without the possibility of hyper-linking the plots, where one actually wants to refer to the respective plots by label.

Reply:
Isn't your second use case dangerous, since it would make the reference only "work" in a single tool, since each tool is free to decide the label appearance?
I can see the argument with future Modelica URIs.

Reply:
Yes, I am aware of that argument against the second case.  It will work in a context where the tool behavior is known, or the figure appearance has been locked by rendering it to a graphics format.  I see that this could be addressed by trying to agree on a labeling scheme, but I'd like to avoid it if we can still reach agreement.

Reply:
I'm personally fine with the current text given your motivation, but I would also be fine with my initial suggestion, knowing that it would limit the possibilities with Modelica URIs.

I do not want the specification to specify the labeling appearance, I think that would be to restrictive.

Reply:
I assume @henrikt-ma means that the user reference using the identifier, but the tool displays it using the label, so it should be safe?

Reply:
I'm not sure I understand the question, but I think we are talking about the potential issue resulting from a user referencing the plot by its label (because the reference is made in an external context where the identifier isn't accessible).  If the same figure is then re-rendered using a different tool (or the same tool, but with different setting for the label scheme), the hard-coded label in the reference could (would) be left dangling.  So: not safe, but not important enough to motivate sacrificing the freedom to choose scheme.